src/operator/tensor/init_op.h

```diff
@@ -732,11 +732,20 @@ inline bool RangeShape(const nnvm::NodeAttrs& attrs,
 }
 
 struct linspace_fwd {
-  template<typename DType>
+  template<typename DType,
+           typename std::enable_if<!std::is_integral<DType>::value, bool>::type = true>
   MSHADOW_XINLINE static void Map(index_t i, double start, double stop, double step,
                                   int req, DType* out) {
     KERNEL_ASSIGN(out[i], req, static_cast<DType>(start + step * i));
   }
+
+  // Floor for int dtype
+  template<typename DType,
+           typename std::enable_if<std::is_integral<DType>::value, bool>::type = true>
+  MSHADOW_XINLINE static void Map(index_t i, double start, double stop, double step,
+                                  int req, DType* out) {
+    KERNEL_ASSIGN(out[i], req, static_cast<DType>(std::floor(start + step * i)));
+  }
 };
```

Comment:
Shall we also add some unittest to tests this new behavior?

Reply:
We already have the tests for negative int:)